### PR TITLE
feat(ui): SidebarConfig.Prepend + poll the sortable 409 e2e tests (#405, #398)

### LINKED ISSUE #398: flake: TestSortable_409InvariantMessage reads the live region after a fixed sleep
`TestSortable_409InvariantMessage` in `core-ui/runtime/sortablelist_e2e_test.go` reads the live region after a fixed `chromedp.Sleep(500*time.Millisecond)`. Under CI load the 409 round trip can take longer than that, so the assertion reads the move announcement that precedes it and fails.

Seen on PR #396, run 34013830009, job `browser e2e · core-ui-runtime`:

```
--- FAIL: TestSortable_409InvariantMessage (1.66s)
    sortablelist_e2e_test.go:739: live region should announce the invariant message, got "Moved to Column B, position 2."
```

The same commit passed the test three times in a row locally and passed on rerun. Neither PR in flight touched the sortable list; the file was last changed in the de-slop sweep.

Fix: replace the sleep with a poll on the live region (`chromedp.Poll` until the text contains the expected message, with a bounded timeout), the shape the rtc and ws suites already use. `TestSortable_409ConflictMessage` next to it has the same sleep and should move with it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01EWVRaHff6UKTY9peDxvazw

### LINKED ISSUE #405: ui: no way to put content above the nav in a MountSidebar drawer body
## The gap

`ui.MountSidebar` mounts a drawer whose body is fixed to the sidebar nav (`sidebarDrawerSlot` renders `sidebarBody(cfg, ...)` and nothing else). A project that wants content **above** the nav inside the mobile drawer has no supported path:

- `SidebarConfig.Footer` renders below the nav, not above.
- `SidebarConfig.ExtraAttrs` lands on the root element, not the drawer body.
- `MountSidebar` hardcodes its own `Slot("body", sidebarDrawerSlot{cfg})`.

## What it blocks

A docs site where the phone header hides the section tabs. The drawer becomes the only navigation, so it needs a section `<select>` at the top (jump between top-level sections) above the nav tree. Today that means hand-rolling the mount:

```go
b := preset.Drawer(name).Hidden().Slot("body", myBody{cfg}) // duplicates MountSidebar
def := b.Build()
mount.MountWidget(&def)
```

...which forks the 8-line mount and risks drifting from `MountSidebar` (page scoping, defaults, future changes).

## Proposal

Add a `Prepend render.HTML` field to `SidebarConfig`, rendered between the title and the `<nav>` in `sidebarBody` (so it appears in the inline sidebar, `SidebarBody`, and the drawer body alike). Empty field, zero markup, same shape as `Footer` today.

Alternative if inline sidebars should not carry it: a `MountSidebarWithBody(r, cfg, body component.Component)` variant.

Either keeps `MountSidebar` as the one-line path for the common case and removes the need to copy it.

### PR DESCRIPTION
**Pull request by DonaldMurillo**
Two open issues in one PR to save a CI run.

## #405 — `SidebarConfig.Prepend`

`MountSidebar` hardcodes the drawer body to the nav, so a docs site whose phone header hides its section tabs had no supported way to put a section `<select>` above the nav tree in the drawer. The only path was to copy the mount and drift from it.

- New `Prepend render.HTML` field, rendered between the title and the `<nav>` in `sidebarBody`, so the inline column, `SidebarBody`, and the drawer all carry it. Empty emits no markup.
- CSS: `.ui-sidebar__prepend` hides with the title and footer in the collapsed rail and the auto-hide rest state.
- Tests: position between title and nav on the inline and body paths, presence in the drawer slot, hide rules in the stylesheet.
- Docs: `ui-new-components.md` sidebar knob list, changelog Added.

Verified with rendered screenshots of the inline column, the collapsed rail (select hidden, rail intact), and the drawer body. The Meridian visual canary passes; its app sidebar is unchanged.

## #398 — sortable 409 e2e flake

`TestSortable_409InvariantMessage` slept 500ms and then read the live region. Under CI load the 409 round trip landed later and the test read the preceding move announcement.

- The 409 family now polls with a bounded 10s timeout: message tests wait for their announcement, fallback tests for the generic copy, DOM-shape tests for the refreshed column or the rollback announcement.
- The runtime announces only after the refresh replaced the DOM, so the announcement also proves the refresh landed. A leaked or wrong message never matches and the poll times out into a failure.
- Proven by delaying the 409 response 1.5s: the sleep-based test failed with the exact CI message, the polled family passed. Full `TestSortable` suite green afterwards.

Sleeps in the non-409 sortable tests (pure client-side moves, no network round trip) are untouched.

Closes #405
Closes #398

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01UbQihscvruKasD9fXthoHs

## Summary by CodeRabbit

* **New Features**
  * Added an optional sidebar content area displayed between the title and navigation.
  * Supported across inline, body, and drawer sidebar layouts.
  * Content can adapt to the current request and hides with the title in collapsed and auto-hide states.
* **Documentation**
  * Documented sidebar content placement, behavior, and footer positioning.
* **Bug Fixes**
  * Improved sortable-list conflict handling and update announcements for more reliable results.

### CODE REVIEW

File: core-ui/runtime/sortablelist_e2e_test.go [hunk 1]
```diff
@@ -519,7 +521,7 @@ func TestSortable_ConflictRefreshEmptyColumn(t *testing.T) {
 		chromedp.Navigate(base+"/"),
 		chromedp.WaitVisible(`#ready`, chromedp.ByID),
 		kbCrossMove("k1"),
-		chromedp.Sleep(500*time.Millisecond),
+		settled(`document.querySelectorAll('[data-fui-sortable-container="b"] [data-fui-sortable-item]').length === 0`),
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply core-ui/runtime/sortablelist_e2e_test.go]: Accepted. Column B starts empty and a rollback also leaves it empty with k1 back in A, so the DOM poll could not tell reconciliation from a revert. The test now waits for the refresh-path announcement ("List refreshed") before reading the column. Proven: with the conflict fetch returning 500 the runtime reverts, the poll times out, and the test fails. Fixed in 77408e79.

File: framework/ui/sidebar_css_test.go [hunk 2]
```diff
@@ -83,3 +83,18 @@ func TestCalloutHiddenAttributeWins(t *testing.T) {
...
+// The collapsed rail and the auto-hide rest state hide the title and
+// footer; Prepend is chrome of the same kind and must hide with them,
+// or a section <select> would poke out of a 64px rail (#405).
+func TestSidebarPrependHidesWithTitleAndFooter(t *testing.T) {
+	css := sidebarCSS(style.Theme{})
+	for _, state := range []string{
+		`[data-collapsed="true"] .ui-sidebar__prepend`,
+		`.ui-sidebar--auto-hide:not(:hover):not(:focus-within) .ui-sidebar__prepend`,
+	} {
+		if !strings.Contains(css, state) {
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply framework/ui/sidebar_css_test.go]: Accepted in part. The test now locates the rule that follows each selector and asserts it sets display:none, so a removed or emptied declaration fails it; flipping the collapsed rule to display:block was the mutation that proved it. Fixed in 77408e79.

Not adding a permanent screenshot test: the collapsed and auto-hide hide rules for the title and footer next to this one carry none either, and the rendered behaviour was verified in this round by reading screenshots of the inline column, the collapsed rail (select hidden, rail intact), and the drawer body. The Meridian visual canary passed unchanged.

**Comment by DonaldMurillo** [comment]
Re the outside-diff finding on `framework/ui/sidebar.go` (auto-hide reveal rule missing its `{`): accepted and fixed in the latest commit.

Confirmed in Chrome before the fix: the rail stayed at 64px under keyboard focus, and the rest-state rule that follows was swallowed too (a qualified rule's prelude runs to the next `{`), so the title and footer leaked into the rail at rest. Shipped since aa569adff; no example uses the variant.

Three guards, each watched fail on the broken rule:

- `TestAutoHideRailWidensOnFocus` (chromium tag) renders the variant, measures 64px at rest with the title hidden and 220px after focusing a link with the title shown.
- `TestAutoHideVariantShipsRevealCSS` now asserts the selector is followed by ` {` and a block restoring 220px, instead of the selector text alone.
- `TestRegisteredStylesheetsAreWellFormed` walks every registered stylesheet and rejects a declaration outside a block or an unbalanced brace. It fires on the sidebar sheet alone and is quiet on every other registered sheet.
